core/bessctl.cc

```diff
@@ -524,7 +524,14 @@ class BESSControlImpl final : public BESSControl::Service {
   Status ResetWorkers(ServerContext*, const EmptyRequest*,
                       EmptyResponse*) override {
     WorkerPauser wp;
-    bess::global_resume_hooks.clear();
+    auto& hooks = bess::global_resume_hooks;
```

Comment:
It seems that an implicit assumption here is that the lifetime of a registered global hook is ended when `ResetWorkers` is invoked. While this is a reasonable choice (unless we want to add another API), one concern is that now `ResetWorkers` and destroying all modules individually have different behaviors. You may want to move this code block to the end of `destroy_worker`, after checking if `num_workers` has become zero.

Reply:
Good point. The process should be consistent across all forms of worker deletion. Moved.